### PR TITLE
GPII-3573: Adding tests for the QSS navigation

### PR DESCRIPTION
**Pull request by danailbd**

### CODE REVIEW

File: tests/QssTestDefs.js [hunk 1]
```diff
@@ -111,6 +111,75 @@ gpii.tests.qss.testPspAndQssVisibility = function (app, params) {
...
+            "{that}.app.qssWrapper.qss.dialog",
+            {
+                key: key,
+                modifiers: modifiers
+            }
+        ]
+    };
+};
+
+gpii.tests.qss.assertFocusedElementIndex = function (expectedIndex) {
```
Comment (danailbd) [inline tests/QssTestDefs.js:134]: Very nice

File: tests/QssTestDefs.js [hunk 2]
```diff
@@ -111,6 +111,75 @@ gpii.tests.qss.testPspAndQssVisibility = function (app, params) {
...
+    };
+};
+
+var qssSettingsCount = 11;
+
+var navigationSequence = [
+    {
+        func: "{that}.app.tray.events.onTrayIconClicked.fire"
+    },
+    gpii.tests.qss.assertFocusedElementIndex(-1),
```
Comment (danailbd) [inline tests/QssTestDefs.js:162]: Probably a comment might be useful for edge cases such as this one.
Additionally it would be better if `Left` key press when no item is focused is also tested.

File: tests/QssTestDefs.js [hunk 3]
```diff
@@ -111,6 +111,75 @@ gpii.tests.qss.testPspAndQssVisibility = function (app, params) {
...
+        resolve: "jqUnit.assertEquals",
+        resolveArgs: [
+            "The correct button in the QSS is focused",
+            expectedIndex,
+            "{arguments}.0"
+        ]
+    };
+};
+
+var qssSettingsCount = 11;
```
Comment (danailbd) [inline tests/QssTestDefs.js:155]: This looks a bit fragile to me but it is probably ok for the tests as the their settings list is controlled.

**Comment by danailbd** [comment]
👍

File: tests/QssTestDefs.js (hunk 3 above)
Comment (georgitodorov) [reply tests/QssTestDefs.js:155]: Yes, I also thought about this. However, getting the actual number of settings within the test sequence will involve more code which will pollute the test cases themselves. That is why I preferred to have this variable.

**Comment by danailbd** [comment]
Very nice, exhaustive tests
